opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```diff
@@ -148,10 +139,10 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         if (servletRequest.getAttribute(SERVER_SPAN_CONTEXT) != null) {
             chain.doFilter(servletRequest, servletResponse);
         } else if (isTraced(httpRequest, httpResponse)){
-            SpanContext extractedContext = tracer.extract(Format.Builtin.HTTP_HEADERS,
+            SpanContext extractedContext = GlobalTracer.get().extract(Format.Builtin.HTTP_HEADERS,
                     new HttpServletRequestExtractAdapter(httpRequest));
 
-            final Span span = tracer.buildSpan(httpRequest.getMethod())
+            final Span span = GlobalTracer.get().buildSpan(httpRequest.getMethod())
```

Comment:
> Tracing may not work correctly if the GlobalTracer is changed

`GlobalTracer#register` seems to gladly replace the existing instance by a new one: https://github.com/opentracing-contrib/java-globaltracer/blob/master/src/main/java/io/opentracing/contrib/global/GlobalTracer.java#L110

If changing after boot is not wanted, it should be a noop with a warn or error-level message. If it's allowed but not recommended, it should still issue a warn-level message.

> before the process handles its primary transaction workload

+1, but we don't control that. The application might start serving requests even if it hasn't finished booting up.

> That said, a change to GlobalTracer config should never crash the process / NPE

We can control this, though. If a change of tracers is not allowed, then the only change we'd ever see is from `NoopTracer` to a concrete tracer. So, storing the tracer in a local var would suffice to avoid any possible issues around this code.